openlr/paths_connector.cpp

```diff
@@ -0,0 +1,257 @@
+#include "openlr/paths_connector.hpp"
+#include "openlr/helpers.hpp"
+
+#include "base/stl_iterator.hpp"
+
+#include <algorithm>
+#include <map>
+#include <numeric>
+#include <queue>
+#include <tuple>
+
+using namespace std;
+
+namespace openlr
+{
+namespace
+{
+size_t IntersectionLen(Graph::EdgeVector a, Graph::EdgeVector b)
+{
+  sort(begin(a), end(a));
+  sort(begin(b), end(b));
+  return set_intersection(begin(a), end(a), begin(b), end(b), CounterIterator()).GetCount();
+}
+
+bool PrefEqualsSuff(Graph::EdgeVector const & a, Graph::EdgeVector const & b, size_t const len)
+{
+  ASSERT_LESS_OR_EQUAL(len, a.size(), ());
+  ASSERT_LESS_OR_EQUAL(len, b.size(), ());
+
+  if (len == 0)
+    return true;
+
+  auto aIt = end(a) - len;
+  auto bIt = begin(b);
+
+  for (; aIt != end(a) && bIt != end(b); ++aIt, ++bIt)
+  {
+    if (*aIt != *bIt)
+      return false;
+  }
+
+  return true;
+}
+
+// Returns a length of the longest suffix of |a| that matches any prefix of |b|.
+// Neither |a| nor |b| can contain several repetitions of any edge.
+// Returns -1 if |a| intersection |b| is not equal to some suffix of |a| and some prefix of |b|.
+int32_t PathOverlappingLen(Graph::EdgeVector const & a, Graph::EdgeVector const & b)
```

Comment:
Why not to return size_t, where zero means no overlapping?

Reply:
Because currently it's unclear to me what's the difference between 0 and -1 return values.

Reply:
-1 is an error, 0 — no intersection, use Dijkstra.

Reply:
Done.